kml/types_v8mm.hpp

```diff
@@ -0,0 +1,314 @@
+                                  visitor(m_imageUrl, "imageUrl"),
+                                  visitor(m_annotation, "annotation"),
+                                  visitor(m_description, "description"),
+                                  visitor(m_visible, "visible"),
+                                  visitor(m_authorName, "authorName"),
+                                  visitor(m_authorId, "authorId"),
+                                  visitor(m_rating, "rating"),
+                                  visitor(m_reviewsNumber, "reviewsNumber"),
+                                  visitor(m_lastModified, "lastModified"),
+                                  visitor(m_accessRules, "accessRules"),
+                                  visitor(m_tags, "tags"),
+                                  visitor(m_toponyms, "toponyms"),
+                                  visitor(m_languageCodes, "languageCodes"),
+                                  visitor(m_properties, "properties"),
+                                  VISITOR_COLLECTABLE)
+
+  DECLARE_COLLECTABLE(LocalizableStringIndex, m_name, m_annotation, m_description,
+                      m_imageUrl, m_authorName, m_authorId, m_tags, m_toponyms, m_properties)
+
+  bool operator==(CategoryDataV8MM const & data) const
+  {
+    double constexpr kEps = 1e-5;
+    return m_id == data.m_id && m_type == data.m_type && m_name == data.m_name &&
+           m_imageUrl == data.m_imageUrl && m_annotation == data.m_annotation &&
+           m_description == data.m_description && m_visible == data.m_visible &&
+           m_accessRules == data.m_accessRules && m_authorName == data.m_authorName &&
+           m_authorId == data.m_authorId && fabs(m_rating - data.m_rating) < kEps &&
+           m_reviewsNumber == data.m_reviewsNumber && IsEqual(m_lastModified, data.m_lastModified) &&
+           m_tags == data.m_tags && m_toponyms == data.m_toponyms &&
+           m_languageCodes == data.m_languageCodes && m_properties == data.m_properties;
+  }
+
+  CategoryData ConvertToLatestVersion() const
+  {
+    CategoryData data;
+    data.m_id = m_id;
+    data.m_type = m_type;
+    data.m_name = m_name;
+    data.m_imageUrl = m_imageUrl;
+    data.m_annotation = m_annotation;
+    data.m_description = m_description;
+    data.m_visible = m_visible;
+    data.m_authorName = m_authorName;
+    data.m_authorId = m_authorId;
+    data.m_rating = m_rating;
+    data.m_reviewsNumber = m_reviewsNumber;
+    data.m_lastModified = m_lastModified;
+    data.m_accessRules = m_accessRules;
+    data.m_tags = m_tags;
+    data.m_toponyms = m_toponyms;
+    data.m_languageCodes = m_languageCodes;
+    data.m_properties = m_properties;
+    return data;
+  }
+
+  bool operator!=(CategoryDataV8MM const & data) const { return !operator==(data); }
+
+  // Unique id (it will not be serialized in text files).
+  MarkGroupId m_id = kInvalidMarkGroupId;
+  // Compilation's type
+  CompilationType m_type = CompilationType::Category;
```

Comment:
Appeared that V8MM format doesn't have `m_type` field. Because of that we had `m_visibility = false`. Using default `m_type` for all V8MM.

Reply:
IT also make sense to mention in the code.

Reply:
Removed this field completly. Added comment.